lib/reversetunnel/localsite.go

```diff
@@ -283,28 +283,35 @@ func (s *localSite) getConn(params DialParams) (conn net.Conn, useTunnel bool, e
 	}
 	// If server ID matches a node that has self registered itself over the tunnel,
 	// return a connection to that node. Otherwise net.Dial to the target host.
-	conn, err = s.dialTunnel(dreq)
-	if err != nil {
-		if !trace.IsNotFound(err) {
-			return nil, false, trace.Wrap(err)
+	conn, tunnelErr := s.dialTunnel(dreq)
+	if tunnelErr != nil {
```

Comment:
I know this code has been written long before your changes, but it would be helpful to flip the condition and handle the happy tunnel dialing path first:
```go
if tunnelErr == nil {
	// Return a tunnel dialed connection.
	return conn, true, nil
}
// Continue with the error case
```
this will reduce the indent and slightly bump the readability.

Reply:
Agreed, will do.